Configure the Swarm cluster.

sedovandrew opened the pull request:
1. The UI service displays the name of the host on which it is running.
2. Configure the Swarm cluster.

sedovandrew committed:
Add env and host info to UI layout

sedovandrew committed:
Deploy with Docker Swarm in GCP

sedovandrew committed:
Added ENV variable to ui service

sedovandrew committed:
Deploy rules

sedovandrew committed:
Added limits to ui, post and comment services

sedovandrew committed:
Added restart policy

sedovandrew committed:
New ignoring files for git

sedovandrew committed:
Splited docker-compose.yml

The main services are in the `docker-compose.yml` file.
Additional services are in the `docker-compose.infra.yml` file.

Nklya reviewed (approved):
Добрый день.

Андрей, условия дз выполнены, все выглядит ок.

Несколько вопросов:
1. advu - у тебя имя пользователя на докер хабе другое?
2. .env не стоит коммитить я думаю, а сделать что-то типа env.example

Ну и по поводу баш скриптов. У них крайне желателен shebang в начале и вместо конструкции `` лучше использовать $()

sedovandrew commented:
Николай, добрый вечер!

1. Да, у меня на docker hub-е имя advu (https://hub.docker.com/u/advu/). Так исторически сложилось. :-)
2. Я тоже думал над `.env`, но решил его закоммитить. С учётом логина и пароля к Grafana этого делать не стоило. А вот если бы там этой информации не было бы? То есть были бы только номера портов и версий, имело бы смысл его коммитить, или в любом случае это плохая практика?
3. Согласен с тобой на счёт shebang-а! Кстати, я сам больше люблю $(), но тут почему-то использовал ``. :-)

Спасибо за рецензию!

Nklya commented:
Из .env лучше сделать например env.example и в нем не указать все то же, что и сейчас, за исключением например что вместо своего пользователя написать username, а в качестве логина/пароля указать например samleuser/samplepass и в README написать что перед работой нужно скопировать env.example в .env и заполнить нужные поля.

Т.е. env.example это полноценный пример с измененными уникальными/секурными данными

sedovandrew committed:
An example of environment variables.

sedovandrew committed:
Split the docker-compose.yml into two parts.

docker-compose.yml for reddit services
docker-compose.infra.yml for infrastructure

chromko reviewed (approved):
Меня тут все устраивает. думаю Можно вмержить все же)
